tests/basic/upgrade_test.go

```diff
@@ -504,3 +505,198 @@ var _ = Describe("{UpgradeVolumeDriverFromCatalog}", func() {
+					expectedUpgradeTime := 9 * len(node.GetStorageDriverNodes())
+					dash.VerifySafely(durationInMins <= expectedUpgradeTime, true, "Verify volume drive upgrade within expected time")
+					upgradeStatus := "PASS"
+					if durationInMins <= expectedUpgradeTime {
+						log.InfoD("Upgrade successfully completed in %d minutes which is within %d minutes", durationInMins, expectedUpgradeTime)
+					} else {
+						log.Errorf("Upgrade took %d minutes to completed which is greater than expected time %d minutes", durationInMins, expectedUpgradeTime)
+						dash.VerifySafely(durationInMins <= expectedUpgradeTime, true, "Upgrade took more than expected time to complete")
+						upgradeStatus = "FAIL"
+					}
+					//check if all the versions are updated except one node.
+					var count = 0
+					for _, n := range storageNodes {
+						updatedPXVersion, err := Inst().V.GetDriverVersionOnNode(n)
+						if err != nil {
+							log.Warnf("error getting driver version, Err: %v", err)
+						}
+						if updatedPXVersion == currPXVersion {
+							count++
+						}
+					}
+
+					if count == 1 {
+						log.Infof("All nodes are updated except the kvdb node")
+					}
+
+					updatedPXVersion, err := Inst().V.GetDriverVersionOnNode(storageNodes[0])
+					if err != nil {
+						log.Warnf("error getting driver version, Err: %v", err)
+					}
+					isDmthinAfterUpgrade, errDmthinCheck := IsDMthin()
+					dash.VerifyFatal(errDmthinCheck, nil, "verified is setup dmthin after upgrade? ")
+					dash.VerifyFatal(isDmthinBeforeUpgrade, isDmthinAfterUpgrade, "setup type remained same pre and post upgrade")
+					majorVersion := strings.Split(currPXVersion, "-")[0]
+					statsData := make(map[string]string)
+					statsData["numOfNodes"] = fmt.Sprintf("%d", numOfNodes)
+					statsData["fromVersion"] = currPXVersion
+					statsData["toVersion"] = updatedPXVersion
+					statsData["duration"] = fmt.Sprintf("%d mins", durationInMins)
+					statsData["status"] = upgradeStatus
+					dash.UpdateStats("px-upgrade-stats", "px-enterprise", "upgrade", majorVersion, statsData)
+				}()
+			})
+			stepLog = "Remove metadatalabel from one of the non kvdb node so that it can become kvdb node"
+			Step(stepLog, func() {
+				log.InfoD(stepLog)
+				// Get kvdb members before we remove the labels
+				_, err := GetAllKvdbNodes()
+				log.FailOnError(err, "Failed to get kvdb nodes before removing labels")
+
+				storageNodes := node.GetStorageNodes()
+				for _, n := range storageNodes {
+					kvdbFlag, err := CheckIfNodeIsKvdbNode(n)
+					log.FailOnError(err, "Failed to check if node is kvdb node")
+					if !kvdbFlag {
+						err = Inst().S.RemoveLabelOnNode(n, k8s.MetaDataLabel)
+						log.FailOnError(err, "Failed to remove labels to node : %s", n.Name)
+						log.InfoD("Removed label: %v to node: %s", k8s.MetaDataLabel, n.Name)
+						break
+					}
+				}
+
+				// Wait for sometime so that the node becomes kvdb node
+				time.Sleep(30 * time.Second)
+
+				_, err = GetAllKvdbNodes()
```

Comment:
can you pls validate the kvdb nodes , GetAllKVDBNOdes will return list of kvdbnodes available